client/components/select-dropdown/index.jsx

```diff
@@ -55,10 +55,14 @@ class SelectDropdown extends Component {
 		} );
 	}
 
-	componentWillReceiveProps() {
+	componentWillReceiveProps( nextProps ) {
 		if ( this.state.isOpen ) {
 			this.closeDropdown();
 		}
+
+		if ( this.props.initialSelected !== nextProps.initialSelected ) {
```

Comment:
What if the `initialSelected` prop changes after the user has made a selection? Should we retain the user's selection rather than overriding it? I'd be annoyed if I had to make my selection again. It looks like `state.selected` will be undefined until a selection is made, so could we add a check for `typeof this.state.selected === 'undefined'` before we `setState`?